spec/src/main/asciidoc/async.asciidoc

```diff
@@ -21,7 +21,7 @@ It is possible for Rest Client interface methods to be declared asynchronous.  T
 
 === Asynchronous Methods
 
-A method is considered to be asynchronous if the method's return type is `java.util.concurrent.CompletableFuture`.
+A method is considered to be asynchronous if the method's return type is `java.util.concurrent.CompletionStage`.
```

Comment:
Just a thought - could we support all return types that implement CompletionStage, including CompletableFuture? It's not necessary and we can discuss later, I just wanted to know if anybody knows about some drawbacks to doing this...